saleor/tax/tests/test_order_calculations.py

```diff
@@ -665,19 +666,10 @@ def test_use_original_tax_rate_when_tax_class_is_removed_from_order_line(
     )
 
     # when
-    for line in lines:
-        tax_class = line.variant.product.tax_class
-        if tax_class:
-            tax_class.delete()
-        tax_class = line.variant.product.product_type.tax_class
-        if tax_class:
-            tax_class.delete()
-        line.refresh_from_db()
-
-    shipping_tax_class = order.shipping_method.tax_class
-    if shipping_tax_class:
-        shipping_tax_class.delete()
-        order.shipping_method.refresh_from_db()
+    TaxClass.objects.all().delete()
+
+    order.refresh_from_db()
+    lines = order.lines.all()
```

Comment:
IMO this lines should be in '#given' section.

Reply:
It is already there :)  It needs to be re-fetched, as tax-classes were removed. See the diff. Previously we were calling `refresh_from_db()` for each line.